Tests/Observable+MoyaSpec.swift

```diff
@@ -343,6 +343,108 @@ class ObservableMoyaSpec: QuickSpec {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var count: Int?
+                _ = observable.map(Int.self, atKeyPath: "count", using: decoder).subscribe(onNext: { value in
+                    count = value
+                })
+                expect(count).notTo(beNil())
+                expect(count) == 1
+            }
+
+            it("maps data representing a Bool at a key path to a Bool value") {
+                let json: [String: Any] = ["isNew": true] // nested json array
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var isNew: Bool?
+                _ = observable.map(Bool.self, atKeyPath: "isNew", using: decoder).subscribe(onNext: { value in
+                    isNew = value
+                })
+                expect(isNew).notTo(beNil())
+                expect(isNew) == true
+            }
+
+            it("maps data representing an Int at a key path to a Bool value") {
+                let json: [String: Any] = ["isNew": 1] // nested json array
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var isNew: Bool?
+                _ = observable.map(Bool.self, atKeyPath: "isNew", using: decoder).subscribe(onNext: { value in
+                    isNew = value
+                })
+                expect(isNew).notTo(beNil())
+                expect(isNew) == true
+            }
+
+            it("maps data representing a String at a key path to a String value") {
+                let json: [String: Any] = ["description": "Something interesting"] // nested json array
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var description: String?
+                _ = observable.map(String.self, atKeyPath: "description", using: decoder).subscribe(onNext: { value in
+                    description = value
+                })
+                expect(description).notTo(beNil())
+                expect(description) == "Something interesting"
+            }
+
+            it("maps data representing a String at a key path should not to an Int value") {
```

Comment:
I'd rearrange the words for tests that "should not do something", to something similar to:
> it shouldn't map String data to an Int value, at a key path

Ideally we would add a context that says:
```
context("when using key path mapping") {
    it("shouldn't map String data to an Int value")
}
```
And then maybe we could create a `sharedExample` and do not duplicate code for tests with and without a key path.
But it isn't required right now and we could make an issue for improving this in the future :)